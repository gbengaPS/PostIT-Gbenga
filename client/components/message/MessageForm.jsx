import React from 'react';
import PropType from 'prop-types';


/**
 * @description MessageForm component
 *
 * @param {object} props -prop object
 *
 * @returns {jsx} -jsx representation of the component
 */
const MessageForm = props => (
  (
    <div>
        <div className="message-form col s12 " >
        <form method="Post" onSubmit={props.handleSubmit}>
            <div className="">
              <textarea
              placeholder="Enter message here"
              name="messageBody" required="required"
              className="message-input-field col s7 m7"
              onChange={props.handleChange}
              value={props.messageBody} />
            </div>
            <div className="message-priority-field col s4" >
              <select
              name="messagePriority"
              className="browser-default"
              onChange={props.handleChange}>
                <option value="Normal">Normal</option>
                <option value="Urgent">Urgent</option>
                <option value="Critical">Critical</option>
              </select>
            </div>
            <div className="col s1">
              <button className="message-button">
                <i className="material-icons">send</i>
              </button>
            </div>
        </form>
        </div>
    </div>
  )
);
MessageForm.propTypes = {
  handleChange: PropType.func,
  handleSubmit: PropType.func,
  messageBody: PropType.string,
};
export default MessageForm;
