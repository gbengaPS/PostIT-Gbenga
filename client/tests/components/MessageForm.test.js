import React from 'react';
import MessageForm from '../../components/message/MessageForm';

describe('MessageForm component ', () => {
  const props = {
    handleChange: () => {},
    handleSubmit: () => {},
    messageBody: 'this is a message',
  };
  it('should render correctly', () => {
    expect(shallow(<MessageForm { ...props}/>)).toMatchSnapshot();
  });
});
