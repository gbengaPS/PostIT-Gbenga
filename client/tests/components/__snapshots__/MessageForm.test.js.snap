// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm component  should render correctly 1`] = `
<div>
  <div
    className="message-form col s12 "
  >
    <form
      method="Post"
      onSubmit={[Function]}
    >
      <div
        className=""
      >
        <textarea
          className="message-input-field col s7 m7"
          name="messageBody"
          onChange={[Function]}
          placeholder="Enter message here"
          required="required"
          value="this is a message"
        />
      </div>
      <div
        className="message-priority-field col s4"
      >
        <select
          className="browser-default"
          name="messagePriority"
          onChange={[Function]}
        >
          <option
            value="Normal"
          >
            Normal
          </option>
          <option
            value="Urgent"
          >
            Urgent
          </option>
          <option
            value="Critical"
          >
            Critical
          </option>
        </select>
      </div>
      <div
        className="col s1"
      >
        <button
          className="message-button"
        >
          <i
            className="material-icons"
          >
            send
          </i>
        </button>
      </div>
    </form>
  </div>
</div>
`;
